### ISSUE TITLE
Fix support for enums with a wrapped val

### ISSUE THREAD

**Issue by bjornbak** [andyglow/scala-jsonschema#232]
**Describe the bug**
We are have enums with a wrapped val in our codebase the same as AnyFooBar in https://github.com/andyglow/scala-jsonschema/blob/master/api/src/test/scala/com/github/andyglow/jsonschema/SchemaMacroSpec.scala#L329-L331 but when I upgrade from 0.6.2 to 0.7.6 get the error below.

**To Reproduce**

Adding this to AsCirceSpec
```
  property("Check that Schema.asCirce translates AnyFooBar") {
    json.Json
      .schema[List[AnyFooBar]]
      .asCirce(Draft07(id = ""))
      .toString() 
  }
```

reproduces the error:
```
[info] - Check that Schema.asCirce translates AnyFooBar *** FAILED ***
[info]   scala.NotImplementedError: an implementation is missing
[info]   at scala.Predef$.$qmark$qmark$qmark(Predef.scala:288)
[info]   at json.Schema$value$minusclass.jsonType(Schema.scala:553)
[info]   at com.github.andyglow.jsonschema.AsDraftSupport.$anonfun$mkOneOf$2(AsDraftSupport.scala:110)
[info]   at scala.Option.map(Option.scala:230)
[info]   at com.github.andyglow.jsonschema.AsDraftSupport.mkOneOf(AsDraftSupport.scala:110)
[info]   at com.github.andyglow.jsonschema.AsDraftSupport.mkOneOf$(AsDraftSupport.scala:104)
[info]   at com.github.andyglow.jsonschema.AsDraft07.mkOneOf(AsDraft07.scala:8)
[info]   at com.github.andyglow.jsonschema.AsDraftSupport$$anonfun$inferSpecifics$1.applyOrElse(AsDraftSupport.scala:172)
[info]   at com.github.andyglow.jsonschema.AsDraftSupport$$anonfun$inferSpecifics$1.applyOrElse(AsDraftSupport.scala:166)
[info]   at scala.PartialFunction$Lifted.apply(PartialFunction.scala:228)
```

**Expected behavior**

**Actual results**
See "To Reproduce"

**Versions:**
 - jsonschema 0.7.6 but anything 0.6.4+
 - scala 2.12

**Additional context**

**Comment by andyglow** [comment]
@bjornbak thank you for reporting it
its being fixed with #234

**Comment by bjornbak** [comment]
Yes, it working with 0.7.7, thank!
